internal/alloycli/automemlimit_cgroups_unsupported.go

```diff
@@ -0,0 +1,25 @@
+//go:build !linux
+// +build !linux
+
+package alloycli
+
+import (
+	"log/slog"
+	"os"
+	"slices"
+	"strings"
+
+	"github.com/KimMachineGun/automemlimit/memlimit"
+	"github.com/grafana/alloy/internal/runtime/logging"
+)
+
+func applyAutoMemLimit(l *logging.Logger) {
+	// For non-linux builds without cgroups, memlimit will always report an error.
+	// However, if the system experiment is requested, we can use the system memory limit provider.
+	// This logic is similar to https://github.com/KimMachineGun/automemlimit/blob/main/memlimit/experiment.go
+	if v, ok := os.LookupEnv("AUTOMEMLIMIT_EXPERIMENT"); ok {
```

Comment:
IIUC, this requires the user to set `AUTOMEMLIMIT_EXPERIMENT`? But isn't the goal that Alloy does it automatically? 

Btw if we're introducing new env vars, they need to be mentioned here:
https://grafana.com/docs/alloy/latest/reference/cli/environment-variables/

Reply:
I'll add it to the docs. This is an environment variable that is specific to the automemlimit module, and it's marked as experimental there so I didn't think we should enable it for all non-linix OSes in this PR. We can consider that change more broadly, this PR was just focused on retaining current functionality while reducing errors.